### PR TITLE
More recipe work

### PR DESCRIPTION
**Pull request by dkolas**

### CODE REVIEW

File: docs/packages/cookbook/article-tagging.rst [hunk 1]
```diff
@@ -0,0 +1,93 @@
+Auto-Tagging and Querying Articles
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This example package collects plain text files and stores them in Steamship. It tags them by running a
+zero-shot classifier with a set of labels provided by the package user.  The results
+can then be queried by tag with a configurable threshold.
+
+..
+    TODO: Test this once oneai tagger is updated
```
Comment (douglas-reid) [inline docs/packages/cookbook/article-tagging.rst:9]: is this TODO relevant to this example?

File: docs/packages/cookbook/collecting-and-querying-sentiment.rst [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+                plugin_handle="oneai-tagger",
+                instance_handle="my-sentiment-analyzer",
+                config={"skills": "sentiment"}
+            )
+
+        # The config method allows your package to return a class
+        # that defines its required configuration.
+        # See <link> for more details.  This package doesn't have any specific
+        # required configuration, so we return the default Config object.
+        def config_cls(self) -> Type[Config]:
```
Comment (douglas-reid) [inline docs/packages/cookbook/collecting-and-querying-sentiment.rst:39]: is it possible to make `config_cls` an abstract method on `PackageService` ?  That would simplify these examples, making them easier to digest.

File: docs/packages/cookbook/article-tagging.rst [hunk 3]
```diff
@@ -0,0 +1,93 @@
+Auto-Tagging and Querying Articles
```
Comment (douglas-reid) [inline docs/packages/cookbook/article-tagging.rst:1]: worth linking to the related blog post or fuller implementation anywhere as a "Learn More" or something ?

File: docs/packages/cookbook/how-to-extract-outline-from-markdown.rst [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+    from steamship.invocable import Config, create_handler, post, PackageService
+
+
+    class MarkdownOutlinePackage(PackageService):
+
+        # The config method allows your package to return a class
+        # that defines its required configuration.
+        # See <link> for more details.  This package doesn't have any specific
+        # required configuration, so we return the default Config object.
+        def config_cls(self) -> Type[Config]:
```
Comment (douglas-reid) [inline docs/packages/cookbook/how-to-extract-outline-from-markdown.rst:45]: same question as above.

File: docs/packages/cookbook/how-to-extract-outline-from-markdown.rst [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+        # The name in the annotation defines the HTTP route suffix, see <link>
+        @post("create_markdown_outline")
+        def create_markdown_outline(self, content: str = None) -> str:
+            """Accept markdown content and extract its outline"""
+
+            # Upload the content of the Markdown file into Steamship.
+            file = File.create(self.client, content, mime_type=MimeTypes.MKD)
+
+            # Now blockify it (convert it to raw text with tags) with the markdown blockifier
+            blockifier = self.client.use_plugin("markdown-blockifier-default")
```
Comment (douglas-reid) [inline docs/packages/cookbook/how-to-extract-outline-from-markdown.rst:60]: would it be too much to mention why `instance_handle` isn't used here?

File: docs/packages/cookbook/article-tagging.rst [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+                instance_handle="my-classifier",
+                config={"tag_kind": "tags",  # The tag.kind we want in the output
+                        "labels": self.config.labels,  # The labels we want to apply
+                        "multi_label": True  # multi-class classification
+                        }
+            )
+
+        # The config method allows your package to return a class
+        # that defines its required configuration.
+        # See <link> for more details.  This package doesn't have any specific
```
Comment (douglas-reid) [inline docs/packages/cookbook/article-tagging.rst:44]: does `<link>` get filled in via some other mechanism?

File: docs/packages/cookbook/article-tagging.rst [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+        # required configuration, so we return the default Config object.
+        def config_cls(self) -> Type[Config]:
+            """Return Config if your package requires no config."""
+            return self.ArticleTaggerConfig
+
+        # This method defines the package user's endpoint for adding content
+        # The @post annotation automatically makes the method available as
+        # an HTTP Post request. The name in the annotation defines the HTTP
+        # route suffix, see <link>
```
Comment (douglas-reid) [inline docs/packages/cookbook/article-tagging.rst:53]: same question re: `<link>`. (and do we want to end with  ' .' ?)

File: docs/packages/cookbook/article-tagging.rst (hunk 1 above)
Comment (dkolas) [reply docs/packages/cookbook/article-tagging.rst:9]: it is not.  Thanks for noticing

File: docs/packages/cookbook/how-to-extract-outline-from-markdown.rst (hunk 5 above)
Comment (dkolas) [reply docs/packages/cookbook/how-to-extract-outline-from-markdown.rst:60]: Rather than doing that, probably should add it

File: docs/packages/cookbook/article-tagging.rst (hunk 6 above)
Comment (dkolas) [reply docs/packages/cookbook/article-tagging.rst:44]: No, I just didn't figure out how to do Sphinx relative links from inside a code block.  A quick googling leads me to believe you can't do that without css hackery, so I'm probably just going to use words for now.

File: docs/packages/cookbook/collecting-and-querying-sentiment.rst (hunk 2 above)
Comment (dkolas) [reply docs/packages/cookbook/collecting-and-querying-sentiment.rst:40]: I don't know the answer to this.  Python's abstract class stuff is still a little mystifying to me.  @EniasCailliau Thoughts?
